contracts/deps/VipCappedGuestListBbtcUpgradeable.sol

```diff
@@ -48,48 +53,37 @@ contract VipCappedGuestListWrapperUpgradeable is OwnableUpgradeable {
         __Ownable_init();
         wrapper = wrapper_;
     }
+    
+    function setWrapper(address wrapper_) public onlyOwner {
+        wrapper = wrapper_;
+    }
 
     /**
      * @notice Invite guests or kick them from the party.
      * @param _guests The guests to add or update.
      * @param _invited A flag for each guest at the matching index, inviting or
      * uninviting the guest.
      */
-    function setGuests(address[] calldata _guests, bool[] calldata _invited)
-        external
-        onlyOwner
-    {
+    function setGuests(address[] calldata _guests, bool[] calldata _invited) external onlyOwner {
         _setGuests(_guests, _invited);
     }
 
-    function vaultBalance(address account) public view returns (uint256) {
-        return BadgerWrapperAPI(wrapper).totalVaultBalance(account);
-    }
-
-    function wrapperBalance(address user) public view returns (uint256) {
-        return BadgerWrapperAPI(wrapper).totalWrapperBalance(user);
-    }
-
     function remainingTotalDepositAllowed() public view returns (uint256) {
-        return totalDepositCap.sub(vaultBalance(wrapper));
+        uint256 totalDeposited = ISett(wrapper).totalSupply().mul(ISett(wrapper).getPricePerFullShare()).div(BASE);
+        return totalDepositCap.sub(totalDeposited);
     }
 
-    function remainingUserDepositAllowed(address user)
-        public
-        view
-        returns (uint256)
-    {
-        return userDepositCap.sub(wrapperBalance(user));
+    function remainingUserDepositAllowed(address user) public view returns (uint256) {
+        uint256 deposited = ISett(wrapper).balanceOf(user).mul(ISett(wrapper).getPricePerFullShare()).div(BASE);
+        return userDepositCap.sub(deposited);
```

Comment:
@sajanrajdev is this logic used to count the exact BADGER deposited instead of remBADGER?

Reply:
Correct, the guestlist will be enabled on remBADGER for a single user to deposit and exact amount of BADGER